xapian-core/include/xapian/weight.h

```diff
@@ -1082,6 +1082,79 @@ class XAPIAN_VISIBILITY_DEFAULT PL2Weight : public Weight {
     double get_maxextra() const;
 };
 
+/// Xapian::Weight subclass implementing the PL2+ probabilistic formula.
+class XAPIAN_VISIBILITY_DEFAULT PL2PlusWeight : public PL2Weight {
```

Comment:
I don't think we want to inherit from `PL2Weight` here - while conceptually there's a relationship between the weighting schemes, it isn't necessary to reflect that in the inheritance tree, as there isn't a situation in which we want to use a `PL2PlusWeight` reference or pointer where a `PL2Weight` reference or pointer is needed.  We always use a pointer or reference to `Xapian::Weight` where polymorphism is needed.

And in practical terms, inheriting like this means that a `PL2PlusWeight` object is about twice the size it needs to be as it has all the member variables from `PL2Weight` (which aren't used) plus its own, and both constructors will get called, and we will call `need_stat()` for the default `PL2Weight` parameters as well as for the actual `PL2PlusWeight` parameters, which may end up with us doing work to provide stats which are never used.

Looking at the BM25+ patch, I see that has the same issue - sorry for missing it there.